### PR TITLE
Fetch messages by state

### PR DESCRIPTION
**Pull request by patrixr**
- Allow the endpoint to support multiple order_ids / item_ids / order_ids e.g `GET /messages?order_id=123,654,789`
- Allow fetching the user's message by state e.g `GET /messages?order_id=123,654,789&state=unread`

### CODE REVIEW

**Comment by steveyken** [comment]
If you send params as an array then rails will automatically parse it for you. E.g. order_id[]=1&order_id[]=2.

Not sure if that is better but is standard for checkboxes etc.

**Comment by patrixr** [comment]
> If you send params as an array then rails will automatically parse it for you. E.g. order_id[]=1&order_id[]=2.
> 
> Not sure if that is better but is standard for checkboxes etc.

oh nice, so I might not need this at all

**Comment by patrixr** [comment]
> > If you send params as an array then rails will automatically parse it for you. E.g. order_id[]=1&order_id[]=2.
> > Not sure if that is better but is standard for checkboxes etc.
> 
> oh nice, so I might not need this at all

@steveyken  I tried that but it didn't seem to work, I'll keep the `split` for the time being and I'll try to adjust later

File: spec/controllers/api/v1/messages_controller_spec.rb [hunk 1]
```diff
@@ -28,20 +32,55 @@
...
+        3.times { create :message, order: order }
+        3.times { create :message, order: order2 }
+        get :index, order_id: "#{order.id},#{order2.id}"
+        expect(subject['messages'].length).to eq(6)
+      end
+
+      it "for a certain state" do
+        3.times { create :message, offer: offer }
+        3.times { create :message, offer: offer, sender_id: user.id }
+        3.times { create :message, offer: offer2 }
```
Comment (namrataukirde) [inline spec/controllers/api/v1/messages_controller_spec.rb]: Which messages have read status here?

File: app/controllers/api/v1/messages_controller.rb [hunk 2]
```diff
@@ -29,15 +29,17 @@ class MessagesController < Api::V1::ApiController
...
+      param :state, String, desc: "Message state (unread|read) to filter on"
       def index
         @messages = @messages.where(id: params[:ids].split(",")) if params[:ids].present?
-        @messages = @messages.where(offer_id: params[:offer_id]) if params[:offer_id].present?
-        @messages = @messages.where(order_id: params[:order_id]) if params[:order_id].present?
-        @messages = @messages.where(item_id: params[:item_id]) if params[:item_id].present?
+        @messages = @messages.where(offer_id: params[:offer_id].split(",")) if params[:offer_id].present?
+        @messages = @messages.where(order_id: params[:order_id].split(",")) if params[:order_id].present?
+        @messages = @messages.where(item_id: params[:item_id].split(",")) if params[:item_id].present?
+        @messages = @messages.with_user_read_state(User.current_user, params[:state]) if params[:state].present?
```
Comment (abhinav098) [inline app/controllers/api/v1/messages_controller.rb]: @patrixr, could you think of any other name please ? 
I think `with_user_read_state` is ambiguous. With user read status is implying that the `read` messages should come.
But instead we are fetching messages here on the based of state which is coming in params.

Thanks

cc- @namrataukirde

File: spec/controllers/api/v1/messages_controller_spec.rb (hunk 1 above)
Comment (patrixr) [reply spec/controllers/api/v1/messages_controller_spec.rb]: the ones created by the user are marked as read by default

**Comment by patrixr** [comment]
> left a review comment.

@abhinav098 i've renamed it

**Comment by abhinav098** [comment]
👍

**Pull request merged**